src/os/bluestore/BlueStore.cc

```diff
@@ -3926,6 +3926,7 @@ void BlueStore::MempoolThread::_resize_shards(bool interval_stats)
 
   for (auto i : store->onode_cache_shards) {
     i->set_max(max_shard_onodes);
+    i->trim();
```

Comment:
Let's move this into set_max itself (and possibly add logic to only do the trim if the max value has dropped by some threshold, but this could be a little tricky).

Reply:
@markhpc Do you think this should apply to all caches that inherit from CacheShard, or just to OnodeCacheShard ?